packages/grid/x-data-grid/src/hooks/features/filter/gridFilterUtils.ts

```diff
@@ -263,40 +259,36 @@ const buildAggregatedFilterItemsApplier = (
 
   // We generate a new function with `eval()` to avoid expensive patterns for JS engines
   // such as a dynamic object assignment, e.g. `{ [dynamicKey]: value }`.
-  const filterItemTemplate = `(function filterItem$$(getRowId, appliers, row, shouldApplyFilter) {
-      ${appliers
-        .map(
-          (applier, i) =>
-            `const shouldApply${i} = !shouldApplyFilter || shouldApplyFilter(${JSON.stringify(
-              applier.item.field,
-            )});`,
-        )
-        .join('\n')}
-
-      const result$$ = {
-      ${appliers
-        .map(
-          (applier, i) =>
-            `${JSON.stringify(String(applier.item.id))}:
-          !shouldApply${i} ?
-            false :
-            appliers[${i}].fn(row),
-      `,
-        )
-        .join('\n')}};
-
-      return result$$;
-    })`;
-
-  const filterItemCore = evalCode<Function>(
-    filterItemTemplate.replaceAll('$$', String(filterItemsApplierId)),
+  const filterItemCore = new Function(
+    'appliers',
+    'row',
+    'shouldApplyFilter',
+    `"use strict";
+${appliers
+  .map(
+    (applier, i) =>
+      `const shouldApply${i} = !shouldApplyFilter || shouldApplyFilter(${JSON.stringify(
+        applier.item.field,
+      )});`,
+  )
+  .join('\n')}
+
+const result$$ = {
+${appliers
+  .map(
+    (applier, i) =>
+      `  ${JSON.stringify(
+        String(applier.item.id),
+      )}: !shouldApply${i} ? false : appliers[${i}].fn(row),`,
+  )
+  .join('\n')}
+};
+
+return result$$;`.replaceAll('$$', String(filterItemsApplierId)),
   );
-  const filterItem: GridFilterItemApplierNotAggregated = (row, shouldApplyItem) => {
-    return filterItemCore(apiRef.current.getRowId, appliers, row, shouldApplyItem);
-  };
   filterItemsApplierId += 1;
 
-  return filterItem;
+  return (row, shouldApplyItem) => filterItemCore(appliers, row, shouldApplyItem);
```

Comment:
Sometimes I like assigning functions to a variable because it gives a better debugging experience, the stack entry would read `filterItem - .../gridFilterUtils.js:294` instead of `(anonymous) - .../gridFilterUtils.js:291`. It's useful when stepping through code but also when collecting performance profiles. With this change I'm also not sure what the name of `filterItemCore` becomes.

It's fine to leave as it is, but just fyi.